### PR TITLE
Mat 8131 urn validation

### PR DESCRIPTION
**Pull request by gregory-akins**
## MADiE PR

Jira Ticket: [MAT-8131](https://jira.cms.gov/browse/MAT-8131)
(Optional) Related Tickets:

### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/typesValidations/fhirR4Validations.test.tsx [hunk 1]
```diff
@@ -71,19 +71,75 @@ describe("Validation Functions", () => {
...
+  });
+
+  it("fails when OID is required && is present but incorrect", async () => {
+    const requiredString = getValidation("uri", true);
+    expect(requiredString).toBeInstanceOf(Yup.StringSchema);
+    await expect(requiredString.validate("urn:oid:1123")).rejects.toThrow(
+      "This is not a valid OID"
+    );
+  });
+  it("succeeds when UUI is present && is not required", () => {
```
Comment (etan-sb) [inline src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/typesValidations/fhirR4Validations.test.tsx]: "succeeds when UUID is present && is not required"

File: src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/typesValidations/fhirR4Validations.ts [hunk 2]
```diff
@@ -138,8 +138,17 @@ export const getDecimalValidator = (required) => {
 };
 
 export const getUriValidator = (required) => {
-  const uriRegex = /\S*/;
-  const baseValidator = Yup.string().matches(uriRegex, "Invalid Uri format");
+  const uriRegex: RegExp = /\S*/;
+  const urnRegex: RegExp = /urn:oid:[0-2](\.(0|[1-9][0-9]*))+/;
+  const baseValidator = Yup.string()
+    .test("urn-specific-test", "This is not a valid OID.", function (value) {
```
Comment (etan-sb) [inline src/components/editMeasure/testCases/components/editTestCase/qiCore/LeftPanel/ElementsTab/builder/element/typesValidations/fhirR4Validations.ts]: Should the helper text be "Invalid OID format" to match the other validator texts?

**Review by etan-sb — APPROVED** [review]
